cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -299,11 +318,13 @@ func minorUpgradePossibleWithPatchRelease(stableVersion, patchVersion *versionut
 	return patchVersion.LessThan(stableVersion)
 }
 
-func getSuggestedEtcdVersion(kubernetesVersion string) string {
-	etcdVersion, err := kubeadmconstants.EtcdSupportedVersion(kubernetesVersion)
+func getSuggestedEtcdVersion(kubernetesVersion string) (string, error) {
+	etcdVersion, warning, err := kubeadmconstants.EtcdSupportedVersion(kubeadmconstants.SupportedEtcdVersion, kubernetesVersion)
 	if err != nil {
-		fmt.Printf("[upgrade/versions] WARNING: No recommended etcd for requested Kubernetes version (%s)\n", kubernetesVersion)
-		return "N/A"
+		return "N/A", errors.Wrap(err, "could not retrieve an etcd version for the target Kubernetes version")
```

Comment:
can we keep getSuggestedEtcdVersion to return only a string like it was before?

```
func getSuggestedEtcdVersion(kubernetesVersion string) (string, error) {
	etcdVersion, warning, err := kubeadmconstants.EtcdSupportedVersion(kubeadmconstants.SupportedEtcdVersion, kubernetesVersion)
	if err != nil {
		klog.Warningf("[upgrade/versions] could not retrieve an etcd version for the target Kubernetes version: %v", err)
		return "N/A"
	}
	if warning != nil {
		klog.Warningf("[upgrade/versions] %v", warning)
	}
	return etcdVersion.String()
}
```

this will keep GetAvailableUpgrades simple.